dev/Repeater/APITests/LayoutTests.cs

```diff
@@ -192,6 +192,33 @@ public void ValidateNonVirtualLayoutDoesNotGetMeasuredForViewportChanges()
             });
         }
 
+        [TestMethod]
+        public void ValidateStackLayoutDoesNotRetainIncorrectMinorWidth()
+        {
+            RunOnUIThread.Execute(() =>
+            {
+                var repeater = new ItemsRepeater();
```

Comment:
Can simplify this a little bit using initializer syntax, and removing the layout part since that is the default. You likely do not need the ScrollViewer for this either.

var repeater = new ItemsRepeater()
{
    ItemsSource = Enumerable.Range(0,1),
    ItemTemplate = ...
};

Content = repeater;
Content.UpdateLayout();

repeater.Measure(new Size(600, 100));
Verify.AreEqual(repeater.ActualSize.X, 600);
repeater.Measure(new Size(300, 100));
Verify.AreEqual(repeater.DesiredSize.X, 300);
repeater.Arrange(new Size(300, 100));
Verify.AreEqual(repeater.ActualSize.X, 300);

The check is also reversed, you want to measure with large size and then a smaller size before the arrange pass.

Reply:
Updated. I kept the scrollviewer for now in order to substitute for the second smaller size measure call before the arrange pass happens (due to the reason I mention [here](https://github.com/microsoft/microsoft-ui-xaml/pull/2072#discussion_r389901958)).